### PR TITLE
Add top user score to timeshift leaderboard

### LINKED ISSUE #9527: Users can not see their own beatmap score and leaderboard position within Timeshift Lobbies
Right now, users can not see their own beatmap score while being in the TImeshift Lobby. They would have to return to the singleplayer map wheel to see their best score, but even then it would be impossible to discern which of the scores displayed in the singleplayer map wheel are currently counting in the timeshift lobby they are playing in.

The only way for users to check their own beatmap score in a Timeshift Lobby is to check the beatmap's individual leaderboard. However, once the leaderboard exceeds 50 players, they can not find their score anymore when they aren't in the top 50. 

Same goes for the overall timeshift lobby leaderboard. They can only see their current leaderboard position when they are within the Top 50 of the lobby's leaderboard. Once they are below the top 50, they can not see their current position, their total score, their amount of completed maps and their amount of play attempts.

The web display of a timeshift lobby does not offer this feature either.

### PR DESCRIPTION
**Pull request by smoogipoo**
Resolves https://github.com/ppy/osu/issues/9527

Prereqs:
- [x] https://github.com/ppy/osu-web/pull/6564

Preview:
image

Position is 0 because of a bug in osu-web: https://github.com/ppy/osu-web/pull/6564#issuecomment-683709821. I've reduced the size of chat a little bit to make a little bit more room, otherwise only one other score could ever be displayed if there's a user top score present.

Diff is chonky but most of it is moving `UserTopScoreContainer` inside `Leaderboard` itself, rather than `BeatmapLeaderboard`.

### CODE REVIEW

File: osu.Game/Online/Leaderboards/Leaderboard.cs
Comment (bdach) [inline osu.Game/Online/Leaderboards/Leaderboard.cs:101]: Although this code was just moved from its original place, it's a bit weird to me how a `null` top score won't persist to the bindable, contrary to everything else... `UserTopScoreContainer` looks to be able to handle nulls ok.

File: osu.Game/Screens/Multi/Match/Components/MatchLeaderboard.cs
Comment (bdach) [inline osu.Game/Screens/Multi/Match/Components/MatchLeaderboard.cs]: Not sure about this coercion to 0 - it doesn't really ever make sense to have a 0 position. Would probably rather prefer just outright throwing, or being able to handle a `null` position and hiding the position text instead.

Comment (smoogipoo) [reply osu.Game/Screens/Multi/Match/Components/MatchLeaderboard.cs]: Sure, the base class already handled nulls.

**Comment by bdach** [comment]
Diff looks good, but I'm holding off approving until the web-side PR gets merged and deployed, since the web-side API change is breaking.

**Review by peppy — APPROVED** [review]

**Pull request merged**
